bika/lims/browser/analysisrequest/publish.py

```diff
@@ -1312,10 +1316,29 @@ def _analyses_data(self, ar, analysis_states=None):
                 andict['previous_results'] = ", ".join(
                     [p['formatted_result'] for p in andict['previous'][-5:]])
 
-            analyses.append(andict)
+            #Append addition analysis for each unit conversion
+            if andict['unit_conversions']:
```

Comment:
This looks like it might fail with a `KeyError`

Reply:
Changed to a get statement